### PR TITLE
Validate contextdir on build

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
We never verified that the context directory passed into the build
command was a valid directory.  When we then slapped a default Containerfile
name onto it, things went south fast if the user had passed us a file and
not a directory.

Fixes: #4383

Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Replaces #4388.   @vrothberg there are no build tests in Podman.  Long story short, it wasn't possible to do once upon a time so we relied on Buildah side.  However, I think given the changes to the test systems that @baude has done, we could do build tests here now.  I didn't include any with this PR, but promise to start trying to add some in the near future, especially in the corner cases where podman build works differently than buildah bud.

**Comment by TomSweeneyRedHat** [comment]
Testing:
```
# podman build file
Error: context must be a directory: /tmp/file

# docker build file
unable to prepare context: context must be a directory: /tmp/file
```

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Could you add the tests, to make sure we don't get a regression.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan I'm not sure I'll get to the tests before early next week, but promise to take a look.  Can we push this through now to cure the issue and I'll pinky swear to the tests?

**Comment by rhatdan** [comment]
Ok open a separate PR.
LGTM
@QiWang19 @nalind @giuseppe @vrothberg PTAL

**Comment by QiWang19** [comment]
LGTM

File: cmd/podman/utils.go [hunk 1]
```diff
@@ -73,3 +73,12 @@ func checkIfFileExists(name string) bool {
 	}
 	return !file.IsDir()
 }
+
+// Check if a file is or is not a directory
+func fileIsDir(name string) bool {
+	file, err := os.Stat(name)
+	if os.IsNotExist(err) {
```
Comment (vrothberg) [inline cmd/podman/utils.go]: I think we need to return and check the error as well. We could face permission errors in which case `file == nil` which could lead to a nil dereference in line 83.

File: cmd/podman/utils.go [hunk 2]
```diff
@@ -73,3 +73,12 @@ func checkIfFileExists(name string) bool {
...
+		return false
+	}
+	return file.IsDir()
```
Comment (rhatdan) [inline cmd/podman/utils.go]: If other error, this will blow up since you will be doing
nil.IsDir()

Comment (TomSweeneyRedHat) [reply cmd/podman/utils.go]: Well darn, I'd thought nil wasn't returned there if there was another file error such as perms, but digging through the source at golang, they always return nil. Fixing and thankx.

**Comment by TomSweeneyRedHat** [comment]
All green now, can we push this through?

**Comment by vrothberg** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
rh-atomic-bot is not supposed to be used on this repo.
I have no idea what it does here.

**Comment by vrothberg** [comment]
> rh-atomic-bot is not supposed to be used on this repo.
> I have no idea what it does here.

I guess it's still activated but it was my fault. Everything build-related lets me believe I'm in the buildah universe :)

**Comment by mheon** [comment]
I think we're stuck, on account of the `homu` job now being present.

**Comment by TomSweeneyRedHat** [comment]
@mheon, is this a close this PR and create a new one moment?

**Comment by mheon** [comment]
I'm pushing the manual merge button.
